index.ts

```diff
@@ -5,55 +5,53 @@ import { ICountry, ICity, IState} from './src/interface';
 export { ICountry, ICity, IState} from './src/interface';
 
  export default {
-	getCountryById: function (id: string): ICountry {
-		return _findEntryById(countryList, id);
-	},
-	getStateById: function (id: string): IState {
-		return _findEntryById(stateList, id);
-	},
-	getCityById: function (id: string): ICity {
-		return _findEntryById(cityList, id);
-	},
-	getStatesOfCountry: function (countryId: string): IState[] {
-		var states = stateList.filter(function (value, index) {
-			return value.country_id === countryId
-		})
-		return states.sort(compare)
-	},
-	getCitiesOfState: function (stateId: string): ICity[] {
-		var cities = cityList.filter(function (value, index) {
-			return value.state_id === stateId
-		})
-		return cities.sort(compare)
-	},
-	getAllCountries: function (): ICountry[] {
-		return countryList;
-	},
-	getCountryByCode: function (code: string): ICountry {
-		return _findEntryByCode(countryList, code);
-	}
-}
-
-let _findEntryById = (source: any, id: string) => {
-	if (id && source != null) {
-		let idx = source.findIndex((c:any) => c.id === id);
-		return (idx !== -1) ? source[idx] : "";
-	}
-	else return "";
+    getStatesOfCountry: function (countryCode: string): IState[] {
+        var states = stateList.filter(function (value) {
+            return value.countryCode === countryCode
+        })
+        return states.sort(compare)
+    },
+    getCitiesOfState: function (countryCode: string, stateCode: string): ICity[] {
+        var cities = cityList.filter(function (value) {
+            return (value.countryCode === countryCode && value.stateCode === stateCode)
+        })
+        return cities.sort(compare)
+    },
+    getCitiesOfCountry: function (countryCode: string): ICity[] {
+        var cities = cityList.filter(function (value) {
+            return (value.countryCode === countryCode)
+        })
+        return cities.sort(compare)
+    },
+    getAllCountries: function (): ICountry[] {
+        return countryList;
+    },
+    getAllStates: function (): IState[] {
+        return stateList;
+    },
+    getAllCities: function (): ICity[] {
+        return cityList;
+    },
+    getCountryByCode: function (isoCode: string): ICountry {
+        return _findEntryByCode(countryList, isoCode);
+    },
+    getStateByCode: function (isoCode: string): IState {
+        return _findEntryByCode(stateList, isoCode);
+    }
```

Comment:
@harpreetkhalsagtbit `getCityByCode(countryCode, stateCode)` is already there. Since we're passing Country Code and State Code, it matches with `getCitiesOfState(countryCode, stateCode)`.

So do we need to replicate?

Reply:
`getCitiesOfState(countryCode, stateCode)` will return array of cities. I want city details with city code as input.

Reply:
@harpreetkhalsagtbit 

We cannot get a single city with `countryCode` and `stateCode` as an input. We need one more identifier to get a single city. For ex: it can be a `cityName`

Reply:
leave it for now..I will check this later....